include/cetl/span.h

```diff
@@ -1 +1,772 @@
+        , size_(source.size())
+    {
+    }
+
+    /// Default copy constructor
+    /// @see std::span::span()
+    constexpr span(const span&) noexcept = default;
+
+    ///@}
+    // +----------------------------------------------------------------------+
+    ///@{ @name Operators
+    // +----------------------------------------------------------------------+
+    /// @copydoc span::operator=()
+    ///
+    constexpr span& operator=(const span& rhs) noexcept = default;
+
+    /// @copydoc span::operator[]()
+    ///
+    constexpr reference operator[](size_type idx) const
+    {
+        CETL_DEBUG_ASSERT((idx < size_), "Indexing outside of the span is undefined.");
+        return data_[idx];
+    }
+
+    ///@}
+    // +----------------------------------------------------------------------+
+    ///@{ @name Iterators
+    // +----------------------------------------------------------------------+
+    /// @copydoc span::begin()
+    ///
+    constexpr iterator begin() const noexcept
+    {
+        return iterator(&data_[0]);
+    }
+    /// @copydoc span::end()
+    ///
+    constexpr iterator end() const noexcept
+    {
+        return iterator(&data_[size_]);
+    }
+    /// @copydoc span::rbegin()
+    ///
+    constexpr reverse_iterator rbegin() const noexcept
+    {
+        return reverse_iterator(end());
+    }
+    /// @copydoc span::rend()
+    ///
+    constexpr reverse_iterator rend() const noexcept
+    {
+        return reverse_iterator(begin());
+    }
+
+    // +----------------------------------------------------------------------+
+    ///@{ @name Element Access
+    // +----------------------------------------------------------------------+
+    /// @copydoc span::front()
+    ///
+    constexpr reference front() const
+    {
+        CETL_DEBUG_ASSERT((size_ > 0), "Calling front on an empty span is undefined.");
+        return *begin();
+    }
+
+    /// @copydoc span::back()
+    ///
+    constexpr reference back() const
+    {
+        CETL_DEBUG_ASSERT((size_ > 0), "Calling back on an empty span is undefined.");
+        return *(end() - 1);
+    }
+
+    /// @copydoc span::data()
+    ///
+    constexpr pointer data() const noexcept
+    {
+        return data_;
+    }
+
+    ///@}
+    // +----------------------------------------------------------------------+
+    ///@{ @name Observers
+    // +----------------------------------------------------------------------+
+
+    ///
+    /// Returns if the span has a zero size or not.
+    /// @return true if the span size is 0.
+    [[nodiscard]] constexpr bool empty() const noexcept
+    {
+        return (size_ == 0);
+    }
+
+    ///
+    /// Returns the size of the span.
+    /// @return The number of elements in the span.
+    /// @see std::span::size
+    constexpr size_type size() const noexcept
+    {
+        return size_;
+    }
+
+    /// @copydoc span::size_bytes()
+    ///
+    constexpr size_type size_bytes() const noexcept
```

Comment:
For this exact reason. Could have had `size` and `count` mean two different things consistently, but no, now we have to qualify size instead.

Reply:
sorry

Reply:
Maybe I'll call my TL, `curmudgeon's template library` instead.

Reply:
`ykgoml::` perhaps? (You Kids! Get Off My Lawn!!)